src/PlayAudioUsingMediaStreams/Controllers/AnimalSoundboardController.cs

```diff
@@ -33,93 +40,108 @@ public async Task Get()
             HttpContext.Response.StatusCode = StatusCodes.Status400BadRequest;
         }
     }
-    
+
     private async Task Soundboard(WebSocket webSocket)
     {
+        string streamSid = null;
         var buffer = new byte[1024 * 4];
-        var receiveResult = await webSocket.ReceiveAsync(
-            new ArraySegment<byte>(buffer), CancellationToken.None);
+        var receiveResult = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+
+        await using var speechRecognitionStream = await _speechRecognitionService.InitStream();
+        while (!receiveResult.CloseStatus.HasValue &&
+               !_applicationLifetime.ApplicationStopping.IsCancellationRequested)
+        {
+            using var jsonDocument = JsonDocument.Parse(Encoding.UTF8.GetString(buffer, 0, receiveResult.Count));
+            var eventMessage = jsonDocument.RootElement.GetProperty("event").GetString();
+
+            switch (eventMessage)
+            {
+                case "connected":
+                    Console.WriteLine("Event: connected");
+                    break;
+                case "start":
+                    Console.WriteLine("Event: start");
+                    streamSid = jsonDocument.RootElement.GetProperty("streamSid").GetString();
+                    Console.WriteLine($"StreamId: {streamSid}");
 
-        var streamSid = string.Empty;
+                    // Do not await task, leave this task running in the background for the duration of the websocket connection
+                    var _ = ListenForSpeechRecognition(webSocket, streamSid, speechRecognitionStream)
```

Comment:
Using the `_` discard to capture the `Task` without actually awaiting it.
This removes warnings that the method isn't awaited.